single Zone/monitor per Task

woodsaj opened the pull request.

mattttt committed:
Redid scaffolding and started the snap page

mattttt committed:
task list pretty much complete

mattttt committed:
Beginnings of the add page. Kinda want to rethink it, but this will get them to launch

woodsaj committed:
wire up new pages

woodsaj committed:
bug fixes

woodsaj commented:
@mattttt this is ready for your review.  The tasks are created as disabled by default as i still need to make some changes to the snap-plugin.

I have had to change the configController so that the NS1 apiKey is stored in clear text in the plugin settings.  This is required as I need to be able to retrieve the key to create new tasks (the key is passed as a configuration option in the task definition).  

The only way to not have to store the key in clear text is to requires users to enter the key on the addTask page everytime they navigate to it.

woodsaj committed:
remove import from module.js

woodsaj committed:
consistant use of ns1Token over ns1_token

woodsaj committed:
update variable name used in headers to ns1Token

woodsaj committed:
set tasks to enabled when creating them

mattttt committed:
Lots of front-end changes to smooth out ux. List page, add page and config page. Still having an issue on Create, will document in PR

mattttt commented:
This is almost ready, but having an issue on the create task. It seems that line 84 of snaptask_add.js isnt being executed until something happens in the interface. In this gif, it'll sit forever, but the instant I click the main menu, it'll redirect me to the list:

![angular-huh](https://cloud.githubusercontent.com/assets/2886187/16212641/2ee8207e-3717-11e6-85a8-6533c64e5717.gif)

Also, when there is only a single snap task left, I'm not able to remove it. 

Once those two things are addressed, I believe this can get pushed up. They have a 9am press release going out, so we either need to push this version or the old one. 

Give me a call anytime tonight when you're awake so we can work it out.

woodsaj committed:
bug fixes issue #15

- Native ES6 promises don't play nicely with angularjs.  Switched
to using $q instead. The create page now redirects to the list page
on successfull creation of the tasks.
- task list was not being updated when the new list was empty. This
resulted in the task list not being updated when the last task was
deleted.
- added some error handling.

woodsaj committed:
add error handling for start/stop tasks.

woodsaj committed:
update version number and update date

woodsaj commented:
@mattttt this is all ready to go.
